src/logging.rs

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+use opentelemetry::{global, trace::TracerProvider};
+use opentelemetry_sdk::trace::{
+    self as sdktrace, Config, Sampler, TracerProvider as SdkTracerProvider,
+};
+use tracing::{event, Level};
+use tracing_opentelemetry::OpenTelemetryLayer;
+use tracing_subscriber::fmt::format::FmtSpan;
+use tracing_subscriber::{
+    fmt, layer::SubscriberExt, EnvFilter, Layer, Registry,
+};
+
+use crate::kvp::EmitKVPLayer;
+
+pub fn initialize_tracing() -> sdktrace::Tracer {
+    let config = Config::default().with_sampler(Sampler::AlwaysOn);
+
+    let provider = SdkTracerProvider::builder().with_config(config).build();
+
+    global::set_tracer_provider(provider.clone());
+    provider.tracer("azure-kvp")
+}
+
+pub fn setup_layers(
+    tracer: sdktrace::Tracer,
+) -> Result<(), Box<dyn std::error::Error>> {
+    let otel_layer = OpenTelemetryLayer::new(tracer);
```

Comment:
You might want to consider applying a filter to this layer, or alternatively moving the current filter that's applied to `stderr_layer` to a [global filter](https://docs.rs/tracing-subscriber/0.3.19/tracing_subscriber/layer/index.html#filtering-with-layers) like this:

```Rust
    let subscriber = Registry::default()
        .with(stderr_layer)
        .with(otel_layer)
        .with(emit_kvp_layer)
        .with(EnvFilter::from_env("AZURE_INIT_LOG"))
```

This works because `EnvFilter` implements both the `Filter` trait and the `Layer` trait. It's important to note that a global filter will stop the other layers from even seeing the event so their filters can only filter from the events allowed through the global filter. This can be good (if you have a lot of events and layers filtering can become expensive!) but also can be confusing if you apply a global filter of say, WARN, and then try to get INFO-level events out of a particular layer with a configurable filter.

Global filters are nice for performance-sensitive applications. I once hit an issue where filtering was consuming something like 10% of the CPU cycles because of a trace-level event inside a loop that executed a _lot_ and every time a half-dozen layers had to say "oh I don't care about this event". [There are even features on tracing to allow you to compile out events](https://github.com/tokio-rs/tracing/blob/b02a700ba6850ad813f77e65144114f866074a8f/tracing/Cargo.toml#L58) for particularly performance-sensitive use-cases.

Since we don't have a ton of events or layers we can comfortably afford to have per-layer filtering, so I lean that direction (even if the OpenTelemetry layer and the stderr layer use the same `Filter`) so we can adjust the KVP filter separately.